core/src/main/java/ar/com/wolox/wolmo/core/permission/PermissionManager.java

```diff
@@ -0,0 +1,89 @@
+package ar.com.wolox.wolmo.core.permission;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Handler;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class PermissionManager {
+
+    private static PermissionManager sInstance;
+
+    private int sRequestCount = 1;
+    private HashMap<Integer, PermissionListener> sRequestListeners = new HashMap<>();
+
+    private PermissionManager() {};
+
+    public synchronized static PermissionManager getInstance() {
+        if(sInstance == null) {
+            sInstance = new PermissionManager();
+        }
+        return sInstance;
+    }
+
+    public boolean requirePermission(Fragment fragment, PermissionListener listener, String... permissions) {
+        String[] ungrantedPermissions = filterUngranted(fragment.getActivity(), permissions);
+        if (ungrantedPermissions.length > 0) {
+            fragment.requestPermissions(ungrantedPermissions, sRequestCount);
+            sRequestListeners.put(sRequestCount++, listener);
+            return false;
+        }
+        listener.onPermissionsGranted();
+        return true;
+    }
+
+    public boolean requirePermission(Activity activity, PermissionListener listener, String... permissions) {
+        String[] ungrantedPermissions = filterUngranted(activity, permissions);
+        if (ungrantedPermissions.length > 0) {
+            ActivityCompat.requestPermissions(activity, ungrantedPermissions, sRequestCount);
+            sRequestListeners.put(sRequestCount++, listener);
+            return false;
+        }
+        listener.onPermissionsGranted();
+        return true;
+    }
+
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, final int[] grantResults) {
+        final PermissionListener listener = sRequestListeners.get(requestCode);
+        if(listener != null) {
+            sRequestListeners.remove(requestCode);
+            if (allGranted(grantResults)) {
```

Comment:
Mmmm, what happens if the user accepts some permissions and rejects others? Like the `Camera` permission and the `Microphone`, which are independent? 
Maybe the callback should report which permissions weren't granted, at least to show an appropriate rationale.